bftengine/include/bftengine/CryptoManager.hpp

```diff
@@ -0,0 +1,76 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License"). You may not use this product except in
+// compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license terms. Your use of
+// these subcomponents is subject to the terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include <memory>
+#include "threshsign/ThresholdSignaturesTypes.h"
+#include "threshsign/IThresholdSigner.h"
+#include "threshsign/IThresholdVerifier.h"
+#include "ReplicaConfig.hpp"
+
+namespace bftEngine {
+
+class CryptoManager {
+ public:
+  static CryptoManager& instance(const ReplicaConfig* config = nullptr, Cryptosystem* cryptoSys = nullptr) {
+    static CryptoManager cm_(config, cryptoSys);
+    return cm_;
+  }
+
+  IThresholdSigner* thresholdSignerForSlowPathCommit() const { return thresholdSigner_.get(); }
+  IThresholdVerifier* thresholdVerifierForSlowPathCommit() const { return thresholdVerifierForSlowPathCommit_.get(); }
+
+  IThresholdSigner* thresholdSignerForCommit() const { return thresholdSigner_.get(); }
+  IThresholdVerifier* thresholdVerifierForCommit() const { return thresholdVerifierForCommit_.get(); }
+
+  IThresholdSigner* thresholdSignerForOptimisticCommit() const { return thresholdSigner_.get(); }
+  IThresholdVerifier* thresholdVerifierForOptimisticCommit() const {
+    return thresholdVerifierForOptimisticCommit_.get();
```

Comment:
If I understand correctly, those pointers can be invalidated on rotation, right ? 
Are they being held by any client of those methods?

Reply:
These pointers are not held by anybody